web-app/src/components/HolidayModal/index.js

```diff
@@ -26,7 +31,14 @@ const HolidayModal = ({
     return true;
   };
 
+  const shouldDisableButtonControl = inputText => {
+    return !inputText.length > 0;
+  };
+
   const duration = getEventDayAmount(holiday);
+  const disableRejectionReasonButton = shouldDisableButtonControl(
+    capturedRejectionReasonText
+  );
```

Comment:
Maybe just me, but I see this as additional complexity that's unneeded.

`disabled={inputText.length === 0}` is still readable and is far less code.